File: typescript/src/PlanetTS1/Model/Player/Buildings/PGasCollector.ts
import { ALocation as ALocation } from "../../Utilities"

export class PGasCollector extends GasCollector {
    loc : ALocation
    use : boolean
    owner : number
    icon : string

    constructor(location : ALocation) {
        super()
        this.loc = location
        this.use = true
        this.owner = 1
        this.icon = "pGC"
    }
}

File: dotnet/src/Model/Player/Buildings/PGasCollector.cs
using System;

namespace PlanetCS1 {
    class PGasCollector : GasCollector {
        public PGasCollector(Location location) {
            this.loc = location;
            this.use = true;
            this.owner = 1;
            this.icon = "pGC";
        }
    }
}